### PR TITLE
add FreeIPA support

### PR DESCRIPTION
**Pull request by tduehr**
Adds support for FreeIPA.

### CODE REVIEW

File: core/src/main/java/org/ldaptive/auth/ext/FreeIPAAuthenticationResponseHandler.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    if (response.getResultCode() != SUCCESS) {
+      final FreeIPAAccountState.Error fError = FreeIPAAccountState.Error.parse(response.getResultCode(), response.getMessage());
+      if (fError != null) {
+        response.setAccountState(new FreeIPAAccountState(fError));
+      }
+    } else if (response.getResult()) {
+      final LdapEntry entry = response.getLdapEntry();
+      final LdapAttribute expTime = entry.getAttribute("krbPasswordExpiration");
+      logger.info("krbPasswordExpiration: {}", expTime);
+      final LdapAttribute loginRemaining = entry.getAttribute("loginGraceRemaining");
```
Comment (dfish3r) [inline core/src/main/java/org/ldaptive/auth/ext/FreeIPAAuthenticationResponseHandler.java]: I can't find any documentation indicating that FreeIPA uses the loginGraceRemaining attribute. Was this a copy paste vestige?

File: core/src/main/java/org/ldaptive/auth/ext/FreeIPAAuthenticationResponseHandler.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+ * @author  tduehr
+ */
+public class FreeIPAAuthenticationResponseHandler implements AuthenticationResponseHandler
+{
+
+  /** Maximum password age. */
+  private int maxPasswordAge = -1;
+
+  /** Maximum password age. */
+  private int maxLoginFailures = -1;
```
Comment (dfish3r) [inline core/src/main/java/org/ldaptive/auth/ext/FreeIPAAuthenticationResponseHandler.java]: There's no way to set this variable. Looks like the variable is needed, I assume this is an oversight.

Comment (tduehr) [inline core/src/main/java/org/ldaptive/auth/ext/FreeIPAAuthenticationResponseHandler.java]: Correct.

File: core/src/main/java/org/ldaptive/auth/ext/FreeIPAAuthenticationResponseHandler.java (hunk 1 above)
Comment (tduehr) [inline core/src/main/java/org/ldaptive/auth/ext/FreeIPAAuthenticationResponseHandler.java]: Yes, sort of. I started with EDirectory and the beginnings of a Kerberos/LDAP AccountState based on the LDAP password management standard. I'll take a look through the code and send another PR to correct these.

Comment (dfish3r) [inline core/src/main/java/org/ldaptive/auth/ext/FreeIPAAuthenticationResponseHandler.java]: Ok. Make sure you review latest rev on master. Thanks.

Comment (tduehr) [inline core/src/main/java/org/ldaptive/auth/ext/FreeIPAAuthenticationResponseHandler.java]: Seems you guys already took care of this. Thanks.

Comment (dfish3r) [inline core/src/main/java/org/ldaptive/auth/ext/FreeIPAAuthenticationResponseHandler.java]: Can you confirm the code in master is working correctly for your FreeIPA instance. We can't test against FreeIPA. Thanks.

Comment (tduehr) [inline core/src/main/java/org/ldaptive/auth/ext/FreeIPAAuthenticationResponseHandler.java]: I have these files in my CAS 4.1 maven overlay and can confirm they work.

**Pull request merged**